app/javascript/Components/Result/left_pane.jsx

```diff
@@ -65,46 +68,48 @@ export class LeftPane extends React.Component {
 
   render() {
     return (
-      <Tabs selectedIndex={this.state.tabIndex} onSelect={tabIndex => this.setState({tabIndex})}>
+      <Tabs
+        selectedIndex={this.getSelectedTabIndex()}
+        onSelect={tabIndex => this.setState({tabIndex})}
+      >
         <TabList>
           <Tab>{I18n.t("activerecord.attributes.submission.submission_files")}</Tab>
           <Tab>{I18n.t("activerecord.models.annotation.other")}</Tab>
-          <Tab disabled={LeftPane.disableTestResultsPanel(this.props)}>
+          <Tab disabled={this.disableTestResultsPanel()}>
             {I18n.t("activerecord.models.test_result.other")}
           </Tab>
-          <Tab disabled={LeftPane.disableFeedbackFilesPanel(this.props)}>
+          <Tab disabled={this.disableFeedbackFilesPanel()}>
             {I18n.t("activerecord.attributes.submission.feedback_files")}
           </Tab>
-          <Tab disabled={LeftPane.disableRemarkPanel(this.props)}>
+          <Tab disabled={this.disableRemarkPanel()}>
             {I18n.t("activerecord.attributes.submission.submitted_remark")}
           </Tab>
         </TabList>
         <TabPanel forceRender={true}>
           <SubmissionFilePanel
             ref={this.submissionFilePanel}
-            result_id={this.props.result_id}
-            submission_id={this.props.submission_id}
-            assignment_id={this.props.assignment_id}
-            grouping_id={this.props.grouping_id}
+            result_id={this.context.result_id}
```

Comment:
I think you can delete these props from `SubmissionFilePanel` altogether. You can keep passing props into `FileViewer` since that's reused in other components outside `Result`.

Reply:
The reason I passed them as props is because they are used in the `componentDidUpdate` method of `SubmissionFilePanel`. The component needs to be able to observe that these values are updated by comparing them to the `prevProps` object. I wasn't able to find a way to make it work without passing these values as props. Here is a stackoverflow answer that I referenced https://stackoverflow.com/questions/53422502/react-context-with-componentdidupdate/54235540#54235540

Reply:
Ah yes, I recall this now from your presentation, thanks! 👍